build.gradle

```diff
@@ -1,63 +1,90 @@
+}
 
-// Apply the java plugin to add support for Java
-apply plugin: 'java'
-apply plugin: 'eclipse'
+version = '0.6.0-SNAPSHOT'
+group 'mt.edu.um.cf2'
 
 // In this section you declare where to find the dependencies of your project
 repositories {
-    // Use 'jcenter' for resolving your dependencies.
-    // You can declare any Maven/Ivy/file repository here.
-    jcenter()
+    mavenCentral()
 }
 
-sourceSets {
-	main {
-		resources.srcDirs = ["src/main/java","src/main/resources"]
-	}
-	test {
-		resources.srcDirs = ["src/test/java", "src/test/resources"]
-	}
-}
+publishing {
+    publications {
+        customLibrary(MavenPublication) {
+            from components.java
+        }
+    }
 
-configurations {
-    extraLibs
+    repositories {
+        maven {
+            name = 'mt.edu.um.cf2.jgribx'
+            url = layout.buildDirectory.dir("repo")
+        }
+    }
 }
 
 test {
-		testLogging {
-    		events 'started', 'passed'
-		}
+    testLogging {
+        events 'started', 'passed'
+    }
+}
+
+configurations {
+    extraLibs
+    runtime
+    ajc
+    aspects
+    expandedJars.transitive = false
+    api {
+        extendsFrom(aspects)
+        extendsFrom(expandedJars)
+    }
+    testArtifacts.extendsFrom testRuntime
+    offlinePluginDependency.extendsFrom(compileOnly)
 }
 
 jar {
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
     from {
-        configurations.extraLibs.collect { it.isDirectory() ? it : zipTree(it) }
+        // Expand all the project jars into the main jar so Raptor can find the CDI annotations
+        (configurations.expandedJars).collect { zipTree(it) }
+    }
+    into('lib') {
+        from((configurations.runtimeClasspath - configurations.offlinePluginDependency) - configurations.expandedJars)
+        exclude 'log4j*'
+        exclude 'slf4j*'
+    }
+
+    manifest {
+        attributes 'Main-Class': 'mt.edu.um.cf2.jgribx.CommandLine'
     }
-	manifest {
-		attributes 'Main-Class': 'mt.edu.um.cf2.jgribx.CommandLine'
-	}
 }
 
 // In this section you declare the dependencies for your production and test code
 dependencies {
     extraLibs(
-        [group: 'commons-cli', name: 'commons-cli', version: '1.4'],
-        ['org.slf4j:slf4j-api:1.7.25'],
-        ['edu.ucar:jj2000:5.2']
+            [group: 'commons-cli', name: 'commons-cli', version: '1.4'],
+            ['org.slf4j:slf4j-api:1.7.25'],
+            ['edu.ucar:jj2000:5.2']
     )
     configurations.implementation.extendsFrom(configurations.extraLibs)
 
+    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.0'
+    
     // Declare the dependency for your favourite test framework you want to use in your tests.
     // TestNG is also supported by the Gradle Test task. Just change the
     // testCompile dependency to testCompile 'org.testng:testng:6.8.1' and add
     // 'test.useTestNG()' to your build script.
-    testImplementation 'junit:junit:4.12'
+    testImplementation 'junit:junit:4.13.1'
 }
```

Comment:
We may want to add a wrapper configuration block at the bottom of the build script and re-run the `wrapper` task. It looks like we only updated the `gradle-wrapper.properties` file when the jar and wrapper scripts should have been updated as well. See [here](https://git.tak.gov/raptor/voltron/novatel/-/blob/develop/build.gradle#L15) for reference, and I'd recommend giving the [wrapper documentation](https://docs.gradle.org/current/userguide/gradle_wrapper.html) a quick read as well if interested.